dbusmock/templates/powerd.py

```diff
@@ -0,0 +1,199 @@
+__email__ = 'jonas.drange@canonical.com'
+__copyright__ = '(c) 2015 Canonical Ltd.'
+__license__ = 'LGPL 3+'
+
+BUS_NAME = 'com.canonical.powerd'
+MAIN_IFACE = 'com.canonical.powerd'
+MAIN_OBJ = '/com/canonical/powerd'
+SYSTEM_BUS = True
+
+
+class SysPowerStates:
+    """States per [1].
+    [1] http://bazaar.launchpad.net/
+        ~phablet-team/powerd/trunk/view/head:/src/powerd.h
+    """
+    POWERD_SYS_STATE_SUSPEND = 0,
+    POWERD_SYS_STATE_ACTIVE = 1
+    POWERD_SYS_STATE_ACTIVE_BLANK_ON_PROXIMITY = 2
+
+    @staticmethod
+    def state_to_string(state):
+        if state == SysPowerStates.POWERD_SYS_STATE_SUSPEND:
+            return 'POWERD_SYS_STATE_SUSPEND'
+        elif state == SysPowerStates.POWERD_SYS_STATE_ACTIVE:
+            return 'POWERD_SYS_STATE_ACTIVE'
+        else:
+            return 'POWERD_SYS_STATE_ACTIVE_BLANK_ON_PROXIMITY'
+
+
+def load(mock, parameters):
+    global _parameters
+    _parameters = parameters
+
+    mock.AddProperties(
+        MAIN_IFACE,
+        dbus.Dictionary({
+            'brightness': _parameters.get('brightness', -1),
+        }, signature='sv'))
+
+    mock._state = SysPowerStates.POWERD_SYS_STATE_ACTIVE
+    mock._state_requests = {}
+    mock._wakeup_requests = {}
+
+
+def wakeup(name, cookie, time):
+    syslog("mock powerd: %s (%s) requesting wakeup at %d, now is %d." % (
+        name, cookie, time, int(timelib.time()))
+    )
+    obj = dbusmock.get_object(MAIN_OBJ)
+    obj.EmitSignal(MAIN_IFACE, 'Wakeup', '', [])
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='si', out_signature='s')
+def requestSysState(self, name, state):
+    cookie = str(uuid.uuid4()).split('-')[0]
+    self._state_requests[cookie] = (name, state)
+    self._state = state
+    syslog("mock powerd: %s (%s) setting sysState to %s." % (
+        name, cookie, SysPowerStates.state_to_string(state)
+    ))
+    self.EmitSignal(MAIN_IFACE, 'SysPowerStateChange', 'i', [state])
+    return cookie
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='s', out_signature='')
+def clearSysState(self, cookie):
+    syslog("mock powerd: %s (%s) cleared %s." % (
+        self._state_requests[cookie][0],
+        cookie,
+        SysPowerStates.state_to_string(self._state_requests[cookie][1])
+    ))
+    del self._state_requests[cookie]
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='st', out_signature='s')
+def requestWakeup(self, name, time):
+    cookie = str(uuid.uuid4()).split('-')[0]
+    syslog("mock powerd: %s (%s) requested wakeup at %d." % (
+        name, cookie, time
+    ))
+    self._wakeup_requests[cookie] = (name, time)
+    t = threading.Timer(
+        time - int(timelib.time()), wakeup, args=(name, cookie, time)
+    )
+    try:
+        t.start()
+    except (KeyboardInterrupt, SystemExit):
+        print("Cancelling foo")
+        t.cancel()
+
+    return cookie
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='s', out_signature='')
+def enableProximityHandling(self, name):
+    pass
+
```

Comment:
Please don't fully spell out these methods if they don't do anything, or only do something trivial. Please rather use `AddMethods()`  and add them wholesale, like the other templates do. This should get rid of most explicitly written methods here.